help/destinations/catalog/crm/microsoft-dynamics-365.md

```diff
@@ -0,0 +1,218 @@
+Refer to the [!DNL Dynamics 365] documentation to [create a field (attribute)](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/customize/create-edit-fields?view=op-9-1) if you need additional guidance.
+
+An example setup within [!DNL Dynamics 365] is shown below:
+![Dynamics 365 UI screenshot showing the custom fields.](../../assets/catalog/crm/microsoft-dynamics-365/dynamics-365-fields.png)
+
+#### Register an application and application user within Azure Active Directory {#prerequisites-app-user}
+
+To enable [!DNL Dynamics 365] to access resources you will need to log in with your [!DNL Azure Account] to [[!DNL Azure Active Directory]](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#register-an-application-with-azure-ad-and-create-a-service-principal) and create the below:
+* Create an [!DNL Azure AD] application
+* Service principal
+* An application secret
+
+You will also need to [create an application user](https://docs.microsoft.com/en-us/power-platform/admin/manage-application-users#create-an-application-user) in [!DNL Azure AD] and associate it with the newly created application.
+
+#### Gather [!DNL Dynamics 365] credentials {#gather-credentials}
+
+Note down the items below before you authenticate to the [!DNL Dynamics 365] CRM destination:
+
+| Credential | Description | Example |
+| --- | --- | --- |
+| `Client ID` | The [!DNL Dynamics 365] Client ID for your [!DNL Azure AD] application. Refer to the [[!DNL Dynamics 365] documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in) for guidance. | `ababbaba-abab-baba-acac-acacacacacac` |
+| `Client Secret` | The [!DNL Dynamics 365] Client Secret for your [!DNL Azure AD] application. You would be using option #2 within the [[!DNL Dynamics 365] documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#authentication-two-options). | `abcde~abcdefghijklmnopqrstuvwxyz12345678` for guidance.|
+| `Tenant ID` | The [!DNL Dynamics 365] Tenant ID for your [!DNL Azure AD] application. Refer to the [[!DNL Dynamics 365] documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in) for guidance.  | `1234567-aaaa-12ab-ba21-1234567890` |
+| `Environment URL` | Refer to the [[!DNL Dynamics 365] documentation](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/developer/org-service/discover-url-organization-organization-service?view=op-9-1) for guidance.| If your [!DNL Dynamics 365] domain is as below, you need the highlighted value.<br> *`org57771b33`.crm.dynamics.com* |
+
+## Guardrails
```

Comment:
Add an anchor tag to Guardrails, please.

Reply:
Oops, great spot.